### ISSUE TITLE
Rotation Mode

### ISSUE THREAD

**Issue by TheGuyWhoBots** [Xian55/WowClassicGrindBot#198]
Hi,

I'm looking for a Rotation mode to be added. I'm having issues with Stealth ability's not going off (Ravage) in AttendGrind mode due to it walking into the mob(breaking stealth). I want full movement control, but bot controls rotation when the conditions are met. I don't think "Druid_Cat_Prowl" works as intended either. I've tried every which way to get it to Ravage when I'm in Prowl (and behind the target of course).

Thank you

**Comment by Xian55** [comment]
Hi thanks for reaching out. 

I see you are having multiple points let me break down.

`Rotation mode` is a great new feature idea. Will look into it in the near future. In the past month I've been working on extending the `RequirementFactory` capabilities to a level where it can provide solid level for a rotation bot. On the other hand put many hours effort to eliminate most of the down time under the hood. Reducing the overall per action delay from ~250ms to around ~10ms - depending on the in-game framerate. Which gives a really high resolution about the current situation and the Goap Planner can make the best decision.

> I want full movement control

This one is a great idea. There's only one limitation. Incase having a keybind on a default movement key like Strafe or any movement key (`Q` or `E` or `WASD`) would break the movement control.

> I'm having issues with Stealth ability's not going off (Ravage)

Can't really do much about it since lacking of target orientation and positional data. The only thing is exposed about the target is the distance from the player but that one also an estimation with 3-5 yard error margin. Reading the memory about the related npc would help but that goes against the whole project principle. 😞 

>  I don't think "Druid_Cat_Prowl" works as intended either.

Will look into it on its own issue later. Checking about if there's a defect. When I worked on the `Form`/`Stance` related feature was mainly focused on Warrior and Paladin, might missed something!

**Comment by Xian55** [comment]
`Druid_Cat_Prowl` should be fixed after #199

**Comment by Xian55** [comment]
After looking at what `ClassConfig.Mode`s are available, with `AttendedGrind` should be able to achieve `Rotation Mode` as the requirements are already met:
* *I want full movement control*
* *bot controls rotation when the conditions are met*

Probably have to expose more conditions for fine tuning, but the base functionality already exists.
